mnefun/_mnefun.py

```diff
@@ -1368,10 +1369,10 @@ def do_preprocessing_combined(p, subjects):
             meg, eeg = 'meg' in raw, 'eeg' in raw
             picks = pick_types(raw.info, meg=meg, eeg=eeg, eog=False,
                                exclude=[])
-            assert type(p.auto_bad_reject) and type(p.auto_bad_flat) == dict
+            flat = p.auto_bad if type(p.auto_bad_flat) == dict else None
```

Comment:
Yes `flat =p.auto_bad_flat if isinstance(p.auto_bad_flat, dict)` will work. I forgot it is set to `None` by default. I don't see where it's set to `float`. 
BTW my syntax checker is flagging `flat =p.auto_bad_flat if isinstance(p.auto_bad_flat, dict)` with `else` expected, is that so?